### PR TITLE
OCPBUGS-5235: Fix mapi_instance_create_failed metric with accelerated networking

### PR DESCRIPTION
**Pull request by RadekManak**
We detect invalid VMsize and Accelerated Networking capability before creating the instance. And not attempt to create it when we know the configuration is invalid. 
We should still increment the mapi_instance_create_failed metric if we are unable to create the machine because of invalid VMsize or when accelerated networking is requested and not supported.

### CODE REVIEW

File: pkg/cloud/azure/actuators/machine/reconciler.go [hunk 1]
```diff
@@ -562,6 +562,12 @@ func (s *Reconciler) createNetworkInterface(ctx context.Context, nicName string)
 
 		skuI, err := s.resourcesSkus.Get(ctx, skuSpec)
 		if err != nil {
+			metrics.RegisterFailedInstanceCreate(&metrics.MachineLabels{
+				Name:      s.scope.Machine.Name,
+				Namespace: s.scope.Machine.Namespace,
+				Reason:    err.Error(),
+			})
+
```
Comment (JoelSpeed) [inline pkg/cloud/azure/actuators/machine/reconciler.go:570]: Do we normally do this for errors or just invalid machine config? Wondering if this should only be in the first condition of the next if statement, WDYT?

Comment (RadekManak) [reply pkg/cloud/azure/actuators/machine/reconciler.go:570]: We increment the metric when we call Azure API with a request needed to create machine that return error e.g. creating machine, IP, interface. We do this for all types of errors. Not just invalid config.

**Comment by RadekManak** [comment]
@sunzhaohua2 Can we get this verified that it solves the issue.

**Comment by JoelSpeed** [comment]
/lgtm

Will add approve once QE have verified

**Pull request merged**